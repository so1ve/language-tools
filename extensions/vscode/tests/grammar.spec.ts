import * as path from 'path';
import * as _fg from 'fast-glob';
import { describe, expect, it } from 'vitest';
import { createGrammarSnapshot } from 'vscode-tmlanguage-snapshot';

const fg: typeof _fg = (_fg as any).default;
const fixturesDir = path.resolve(__dirname, '../../../test-workspace/grammar');
const packageJsonPath = path.resolve(__dirname, '../package.json');

describe('grammar', async () => {
	const snapshot = await createGrammarSnapshot(packageJsonPath);
	const cases = await fg(path.join(fixturesDir, "**").replace(/\\/g, "/"));

	for (const kase of cases) {
		it(path.relative(fixturesDir, kase).replace(/\\/g, "/"), async () => {
			const result = await snapshot(kase);

			expect(result).toMatchSnapshot();
		});
	}
});
