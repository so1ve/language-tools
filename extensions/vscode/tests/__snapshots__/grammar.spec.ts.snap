// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`grammar > basic.vue 1`] = `
"><div></div>
#^ source.vue punctuation.definition.tag.begin.html.vue
# ^^^ source.vue entity.name.tag.div.html.vue
#    ^ source.vue meta.tag-stuff punctuation.definition.tag.end.html.vue
#     ^^ source.vue punctuation.definition.tag.begin.html.vue
#       ^^^ source.vue entity.name.tag.div.html.vue
#          ^ source.vue punctuation.definition.tag.end.html.vue"
`;
